src/arch/riscv/interrupts.cc

```diff
@@ -43,7 +43,7 @@ Interrupts::Interrupts(const Params &p) : BaseInterrupts(p),
     for (uint8_t i = 0;
         i < p.port_local_interrupt_pins_connection_count;
         ++i) {
-            uint8_t interruptID = p.local_interrupt_ids[i];
+            int interruptID = p.local_interrupt_ids[i];
             assert(interruptID >= 0);
```

Comment:
I don't think this is the right fix. `p.local_interrrupts_ids` is a unsigned type.

We should remove the assert.

Reply:
I think it's right to remove the assert. Having taken a moment to understand the error. The compiler test is simply failing stating as it's an unsigned int and therefore `assert(interruptID >=)` is always true.

Reply:
Fixed.